### ISSUE TITLE
Can't drag and drop in 16.0.0 on a shared link.  Worked in 15.x

### ISSUE THREAD

**Issue by evanrich** [nextcloud/server#15319]
It looks like there is an issue with drag and drop with shared links in the latest 16.0 beta.  I upgraded and everything works great, except drag and drop, although only for share links.  I am able to drag and drop fine, but when I select a "shared" link and then try to drag and drop, it just hangs.

### Steps to reproduce
1.  Create a Shared link, allow edit permissions (upload/etc)
2. Open share link (incognito, regular, doesnt matter)
3. Try to drag a file into a folder.

### Expected behaviour
File should drop (worked in latest 15.x release)

### Actual behaviour
The file just hangs over the folder, never drops/moves

### Server configuration

**Operating system**:
Alpine linux (docker container)

**Web server:**

**Database:**
MariaDB 10.x

**PHP version:**
7.2.10

**Nextcloud version:** (see Nextcloud admin page)
16.0.0

**Updated from an older Nextcloud or fresh install:**  Updated

**Signing status:**

```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- files_rightclick
	- EXTRA_FILE
		- README.md

Raw output
==========
Array
(
    [files_rightclick] => Array
        (
            [EXTRA_FILE] => Array
                (
                    [README.md] => Array
                        (
                            [expected] => 
                            [current] => cf73849388838de5037624e53303618579b164ec69fd55834ce4c5332ae33f778839e36298cbcc8ede90620927e076dbb8883479754e4c09b0491b15bf7653f3
                        )

                )

        )

)
```

**List of activated apps:**

```
Enabled:
  - accessibility: 1.2.0
  - activity: 2.9.1
  - admin_audit: 1.6.0
  - bruteforcesettings: 1.3.0
  - camerarawpreviews: 0.6.5
  - checksum: 0.4.2
  - cloud_federation_api: 0.2.0
  - comments: 1.6.0
  - dav: 1.9.2
  - event_update_notification: 0.3.4
  - federatedfilesharing: 1.6.0
  - federation: 1.6.0
  - files: 1.11.0
  - files_pdfviewer: 1.5.0
  - files_rightclick: 0.13.0
  - files_sharing: 1.8.0
  - files_texteditor: 2.8.0
  - files_trashbin: 1.6.0
  - files_versions: 1.9.0
  - files_videoplayer: 1.5.0
  - firstrunwizard: 2.5.0
  - gallery: 18.3.0
  - logreader: 2.1.0
  - lookup_server_connector: 1.4.0
  - metadata: 0.9.0
  - nextcloud_announcements: 1.5.0
  - notifications: 2.4.1
  - oauth2: 1.4.2
  - password_policy: 1.6.0
  - previewgenerator: 2.1.0
  - privacy: 1.0.0
  - provisioning_api: 1.6.0
  - recommendations: 0.4.0
  - registration: 0.4.6
  - serverinfo: 1.6.0
  - sharebymail: 1.6.0
  - sharerenamer: 2.6.0
  - support: 1.0.0
  - survey_client: 1.4.0
  - systemtags: 1.6.0
  - theming: 1.7.0
  - twofactor_backupcodes: 1.5.0
  - unsplash: 1.1.3
  - updatenotification: 1.6.0
  - viewer: 1.0.0
  - workflowengine: 1.6.0
Disabled:
  - calendar
  - caniupdate
  - encryption
  - files_automatedtagging
  - files_downloadactivity
  - files_external
  - socialsharing_email
  - spreed
  - terms_of_service
  - user_external
  - user_ldap
```

**Nextcloud configuration:**

```
I"memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "10.44.0.69:443",
            "cloud.redacted.com",
            "192.168.1.21:443"
        ],
        "overwrite.cli.url": "https:\/\/cloud.evanrichardsonphotography.com",
        "overwritehost": "cloud.evanrichardsonphotography.com",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "16.0.0.9",
        "installed": true,
        "maintenance": false,
        "theme": "",
        "log_type": "file",
        "logfile": "\/data\/nextcloud.log",
        "loglevel": "3",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpsecure": "ssl",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "updater.release.channel": "beta",
        "twofactor_enforced": "false",
        "twofactor_enforced_groups": [],
        "twofactor_enforced_excluded_groups": [],
        "mysql.utf8mb4": true

```

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** No

**Are you using an external user-backend, if yes which one:**No

NExtcloud logs:
```{"reqId":"nfqXzP4xjiFiLK6EJsNR","level":3,"time":"2019-04-30T17:07:55+00:00","remoteAddr":"127.0.0.1","user":"--","app":"jsresourceloader","method":"GET","url":"\/index.php\/s\/nitda20190420","message":"Could not find resource core\/vendor\/marked\/marked.min.js to load","userAgent":"Mozilla\/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit\/605.1.15 (KHTML, like Gecko) Version\/12.1 Mobile\/15E148 Safari\/604.1","version":"16.0.0.9"}
{"reqId":"daRWeF3ZCuF3TL7DzOnB","level":3,"time":"2019-04-30T17:08:33+00:00","remoteAddr":"127.0.0.1","user":"--","app":"jsresourceloader","method":"GET","url":"\/index.php\/s\/nitda20190420?path=%2FFinals","message":"Could not find resource core\/vendor\/marked\/marked.min.js to load","userAgent":"Mozilla\/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit\/605.1.15 (KHTML, like Gecko) Version\/12.1 Mobile\/15E148 Safari\/604.1","version":"16.0.0.9"}
{"reqId":"PB19UUH9VX0SKAQNCVJe","level":3,"time":"2019-04-30T17:08:46+00:00","remoteAddr":"127.0.0.1","user":"--","app":"jsresourceloader","method":"GET","url":"\/index.php\/s\/nitda20190420?path=%2FFinals","message":"Could not find resource core\/vendor\/marked\/marked.min.js to load","userAgent":"Mozilla\/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit\/605.1.15 (KHTML, like Gecko) Version\/12.1 Mobile\/15E148 Safari\/604.1","version":"16.0.0.9"}
{"reqId":"unCYiTG5QY9pHsOC4DMm","level":3,"time":"2019-04-30T17:09:38+00:00","remoteAddr":"127.0.0.1","user":"--","app":"jsresourceloader","method":"GET","url":"\/index.php\/s\/nitda20190420?path=%2FFinals","message":"Could not find resource core\/vendor\/marked\/marked.min.js to load","userAgent":"Mozilla\/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit\/605.1.15 (KHTML, like Gecko) Version\/12.1 Mobile\/15E148 Safari\/604.1","version":"16.0.0.9"}
{"reqId":"7kdQAfnZiJRllG99szSB","level":3,"time":"2019-04-30T20:13:09+00:00","remoteAddr":"127.0.0.1","user":"--","app":"jsresourceloader","method":"GET","url":"\/index.php\/s\/KatMoua","messa
```

**Comment by im-fabian** [comment]
+1 here

**Comment by lopezio** [comment]
+1 here

16.0.0 (release) upgraded from 15.0.7.

Details in JS Console on Firefox (actual domain changed):

<pre>
The resource from “https://cloud.example.com/login” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff).[Learn More]
LE9ipLbmpGGZdKQ
JQMIGRATE: Migrate is installed, version 1.4.1 main.js:278:521
The escapeHTML library is deprecated! It will be removed in nextcloud 19.
main.js:750:50
TypeError: OCA.Files.Semaphore is not a constructor[Learn More] file-upload.js:564:19
Source map error: TypeError: NetworkError when attempting to fetch resource.
Resource URL: https://cloud.example.com/core/js/dist/main.js?v=f89fdba3-20
Source Map URL: main.js.map[Learn More]
TypeError: OCA.Files.Semaphore is not a constructor[Learn More] file-upload.js:564:19
    submitUploads https://cloud.example.com/apps/files/js/file-upload.js?v=f89fdba3-20:564
    onNoConflicts https://cloud.example.com/apps/files/js/file-upload.js?v=f89fdba3-20:992
    checkExistingFiles https://cloud.example.com/apps/files/js/file-upload.js?v=f89fdba3-20:777
    add https://cloud.example.com/apps/files/js/file-upload.js?v=f89fdba3-20:1010
    _trigger https://cloud.example.com/core/js/dist/main.js?v=f89fdba3-20:290
    _onAdd jQuery
    each https://cloud.example.com/core/js/dist/main.js?v=f89fdba3-20:28
    _onAdd jQuery
    i/e.widget/</a[t]</< https://cloud.example.com/core/js/dist/main.js?v=f89fdba3-20:290
    _onDrop jQuery
    u https://cloud.example.com/core/js/dist/main.js?v=f89fdba3-20:39
    add https://cloud.example.com/core/js/dist/main.js?v=f89fdba3-20:39
    always https://cloud.example.com/core/js/dist/main.js?v=f89fdba3-20:39
    _onDrop jQuery
    i/e.widget/</a[t]</< https://cloud.example.com/core/js/dist/main.js?v=f89fdba3-20:290
    a https://cloud.example.com/core/js/dist/main.js?v=f89fdba3-20:290
    dispatch https://cloud.example.com/core/js/dist/main.js?v=f89fdba3-20:39
    handle https://cloud.example.com/core/js/dist/main.js?v=f89fdba3-20:39
</pre>

**Comment by lopezio** [comment]
Fix proposed in related/duplicate issue   https://github.com/nextcloud/server/issues/14986#issuecomment-484451455 fixes the issue for me.

(copying fix here):

Adding 

`\OCP\Util::addScript('files', 'merged-index');` 

on line 415 of 

`apps/files_sharing/lib/Controller/ShareController.php` 

Hopefully fix will be released soon

**Comment by jospoortvliet** [comment]
Fix targetted for 16.0.1! Let's track this in #14986
